services/blue/src/api/Handlers/ConfirmSelectedRoomTypeHandler.cs

```diff
@@ -30,9 +29,9 @@ public async Task Handle(ConfirmSelectedRoomType message, IMessageHandlerContext
                 // TODO: reply to caller with "fault" message.
                 throw new Exception($"Could not find pending reservation for {nameof(message.OrderId)} {message.OrderId}");
 
-            var pendindReservationIsValid = await IsPendingReservationValidAsync(connection, pendingReservation);
+            var isValid = await IsPendingReservationValidAsync(connection, pendingReservation);
```

Comment:
Add .ConfigureAwait(false) => everywhere where an await is.

```suggestion
            var isValid = await IsPendingReservationValidAsync(connection, pendingReservation).ConfigureAwait(false);
```

Reply:
Ok in libraries, handlers don't need it.
It also wouldn't allow this syntax:
```csharp
await using var connection = await _connectionFactory.CreateAsync();
```

Reply:
Why don't we need it in handlers? In most samples and the docs from particular they use it even in handlers.

Why wouldn't you allow this syntax? Too confusing with two awaits? How would you write it then, because that way it's clear that we are using an IAsyncDisposable.

Reply:
Handlers don't need it since they're running in a console app, which doesn't have a `SynchronizationContext` by default.

Using `ConfigureAwait(false)` in libraries (e.g. our `infrastructure`/`ITOps`) is best practice, since they're meant to be used in another, unknown environment (e.g. .NET Framework / ASP.NET MVC). At the moment, not even ASP.NET Core has a `SynchronizationContext`.

If we enforced the use of `ConfigureAwait` in handlers, it would add unnecessary complexity in the code and it wouldn't allow to write (imo) clean syntax like: 
```csharp
await using var connection = await _connectionFactory.CreateAsync();
``` 
it would instead be:
```csharp
var connection = await _connectionFactory.CreateAsync().ConfigureAwait(false);
await using (connection.ConfigureAwait(false))
{
}
```

Reply:
Ah yes you're right, I forgot that even ASP.Net Core has no SynchronisationContext. Too much WPF the last weeks ^^